### ISSUE TITLE
StringInterpolation is slow and inefficient

### ISSUE THREAD

**Issue by mkeskells** [rorygraves/scalac_perf#24]
String Interpolation is slow at run time and inefficient code generation, with lots of temporary garbage generated

convert s and raw to macros (like f is already) 

generate optimal code

This is mostly a macro writing issue + one line to register the fast path for the macro in FastPath

For the macro structure see FormatInterpolater
Probably should improve that code at  the same time or in a separate issue

maybe revert or test with https://github.com/rorygraves/scalac_perf/issues/14 as an example

**Comment by mkeskells** [comment]
first stab https://github.com/rorygraves/scalac_perf/tree/mike/2.12_x_StringInterpolation
then https://github.com/scala/scala/pull/6093 for intrinsic version to be followed by macro version

**Comment by mkeskells** [comment]
PR https://github.com/scala/scala/pull/6041

**Comment by lrytz** [comment]
https://github.com/scala/scala/pull/6093

**Comment by mkeskells** [comment]
https://github.com/scala/scala/pull/6093 is merged - ned t confirm the 2.13 plan before we close this issue though. May consider opening a similar issue for f"", for some simple cases

**Comment by mkeskells** [comment]
@retronym - has the macro version of this been merged into 2.13.x?

**Comment by retronym** [comment]
Nope. It is tracked as https://github.com/scala/scala-dev/issues/475.
